sky/utils/rich_utils.py

```diff
@@ -1,16 +1,21 @@
 """Rich status spinner utils."""
```

Comment:
This is great! However, since this can affect many other part of the code, let's move this to another PR, and keep this PR clean for only having the parallelism changes.

Reply:
Sure, but since status of sub thread is silent on master, there would be a UX degradation in between, is that okay?